_posts/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins.md

```diff
@@ -0,0 +1,258 @@
+---
+layout: post
+title: "Develop guideline: Multiple Data Source in OpenSearch Dashboards and Plugins"
+authors:
+ - zhyuanqi
+date: 2024-05-01
+categories:
+ - technical-post
+meta_keyword: OpenSearch multiple data source, OpenSearch Dashboards
+meta_description: This blog provides a guideline on howw integrate multiple data source with Dashboards pluginds
+---  
+
+OpenSearch introduced Multiple Data Source feature in 2.4 release. The feature enables user to explore, visualize, discover, manage data by connecting to multiple data sources (self-managed clusters, Amazon OpenSearch Service). Staring 2.14 release, 9 OpenSearch Dashboards plugins integrated to support Multiple Data Sources. Now users can enjoy the benefit of accessing data from remote clusters not only within OpenSearch Dashboards core but also within plugins. This expansion broadens the range of available data sources, enhancing users' capabilities to work with diverse data sources.
+
+For developers seeking to integrate OpenSearch Dashboards plugins with multiple data sources, this blog post aims to offer a concise introduction on how to do it. If you're new to the concept of multiple data sources, consider starting with the article Configuring and using multiple data sources to gain a solid understanding before diving into plugin integration.
+
+
+## 1. Getting Started
+
+As a developer, you can integrate multiple data sources into dashboards using plugins in just a few steps. In this tutorial, we will provide you with the necessary steps and examples to help you successfully implement, configure, and get started using multiple data sources in OpenSearch Dashboards. 
+
+### 1.1 Step 1: Set up local development
+
+Before getting started, ensure you have set up a local development environment for OpenSearch Dashboards following our developer guide.  Once your Dashboards instance is up and running, you can generate your own Dashboards plugin using the OpenSearch Dashboards Plugin Generator by running the CLI command below:
+```
+node scripts/generate_plugin --name my_plugin_name
```

Comment:
Unrelated to this pr. But perhaps we can consider updating the cli to include a question to ask if the developers wants to setup their plugin for osd?

Reply:
I think it includes multiple questions includes what kind of internal plugin and what the part will be. 
```
? Will this plugin be part of the OpenSearch Dashboards repository? Yes
? What type of internal plugin would you like to create (Use arrow keys)
❯ OpenSearch Dashboards Example 
  OpenSearch Dashboards OSS 
  OpenSearch Dashboards OSS Functional Testing 
```